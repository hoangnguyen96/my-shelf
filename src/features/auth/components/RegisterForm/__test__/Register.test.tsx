import { render } from "@testing-library/react";
import { useRouter } from "next/navigation";
import { RegisterForm } from "..";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

describe("Register", () => {
  const mockPush = jest.fn();

  beforeEach(() => {
    (useRouter as jest.Mock).mockReturnValue({
      push: mockPush,
    });
  });

  it("Should render correctly snapshot", () => {
    const { container } = render(<RegisterForm />);

    expect(container).toMatchSnapshot();
  });
});
