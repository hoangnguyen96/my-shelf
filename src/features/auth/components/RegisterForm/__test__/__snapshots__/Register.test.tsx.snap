// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Should render correctly snapshot 1`] = `
<div>
  <form
    class="css-0"
    style="margin-top: 40px;"
  >
    <div
      class="chakra-form-control css-13srdsj"
      role="group"
    >
      <label
        class="chakra-form__label css-uvho5g"
        for="name"
        id="field-:r0:-label"
      >
        Name
      </label>
      <div
        class="chakra-input__group css-bx0blc"
        data-group="true"
      >
        <input
          class="chakra-input css-1x36n8t"
          id="name"
          name="username"
          placeholder="Username..."
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="chakra-form-control css-13srdsj"
      role="group"
    >
      <label
        class="chakra-form__label css-uvho5g"
        for="email"
        id="field-:r1:-label"
      >
        Email
      </label>
      <div
        class="chakra-input__group css-bx0blc"
        data-group="true"
      >
        <input
          class="chakra-input css-1x36n8t"
          id="email"
          name="email"
          placeholder="Email..."
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="chakra-form-control css-13srdsj"
      role="group"
    >
      <label
        class="chakra-form__label css-uvho5g"
        for="password"
        id="field-:r2:-label"
      >
        Password
      </label>
      <div
        class="chakra-input__group css-bx0blc"
        data-group="true"
      >
        <input
          class="chakra-input css-1x36n8t"
          id="password"
          name="password"
          placeholder="Password..."
          type="password"
          value=""
        />
        <div
          class="chakra-input__right-element css-vcnsx7"
        >
          <button
            aria-label="Show password"
            class="chakra-button css-1rwczf8"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-1qpj0zo"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                />
                <path
                  d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="chakra-form-control css-13srdsj"
      role="group"
    >
      <label
        class="chakra-form__label css-uvho5g"
        for="confirmPassword"
        id="field-:r3:-label"
      >
        Confirm Password
      </label>
      <div
        class="chakra-input__group css-bx0blc"
        data-group="true"
      >
        <input
          class="chakra-input css-1x36n8t"
          id="confirmPassword"
          name="confirmPassword"
          placeholder="ConfirmPassword..."
          type="password"
          value=""
        />
        <div
          class="chakra-input__right-element css-vcnsx7"
        >
          <button
            aria-label="Show password"
            class="chakra-button css-1rwczf8"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-1qpj0zo"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                />
                <path
                  d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
    </div>
    <button
      class="chakra-button css-1qhatj0"
      data-testid="submit-register"
      disabled=""
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;
